### PR TITLE
On MacOS, prevent user from launching the app when it is translocated.

### PR DESCRIPTION
**Pull request by AkiSakurai**
The app doesn't work properly with App Translocation [^1].

[^1]: https://lapcatsoftware.com/articles/app-translocation.html

### CODE REVIEW

File: OpenTabletDriver.UX.MacOS/Program.cs [hunk 1]
```diff
@@ -7,6 +8,18 @@ class Program
...
         {
+            var baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+
+            if (baseDirectory.StartsWith("/private/var/folders"))
+            {
+                _ = new Application(Eto.Platforms.Mac64);
+                MessageBox.Show(
+                    "OpenTabletDriver cannot be run from this folder. " +
+                    "Please move it to elsewhere, " +
+                    "such as to the Applications folder.");
```
Comment (vedattt) [inline OpenTabletDriver.UX.MacOS/Program.cs:19]:
```suggestion
                    "OpenTabletDriver cannot be run from this folder (/private/var/folders). " +
                    "Please move it elsewhere, such as to the Applications folder.");
```
Rewording and clarification.

**Review by vedattt — CHANGES_REQUESTED** [review]

File: OpenTabletDriver.UX.MacOS/Program.cs (hunk 1 above)
Comment (AkiSakurai) [reply OpenTabletDriver.UX.MacOS/Program.cs:19]: The path /private/var/folders is invisible to the user.

From the users point of view, the app is still in Downloads folder. 

The system silently copy the app to that folder and launch from there if the app is downloaded from internet.

Comment (vedattt) [reply OpenTabletDriver.UX.MacOS/Program.cs:19]: Ah, I see, I'm not really familiar with macOS.
Probably still worth considering clarifying it without the path though:
```suggestion
                    "OpenTabletDriver cannot be run from this folder for security reasons. " +
                    "Please move it elsewhere, such as to the Applications folder.");
```

**Review by vedattt — APPROVED** [review]

File: OpenTabletDriver.UX.MacOS/Program.cs [hunk 2]
```diff
@@ -7,6 +8,17 @@ class Program
...
         public static void Main(string[] args)
         {
+            var baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+
+            if (baseDirectory.StartsWith("/private/var/folders"))
+            {
+                _ = new Application(Eto.Platforms.Mac64);
+                MessageBox.Show(
+                    "OpenTabletDriver cannot be run from this folder for security reasons. " +
+                    "Please move it elsewhere, such as to the Applications folder.");
```
Comment (InfinityGhost) [inline OpenTabletDriver.UX.MacOS/Program.cs:18]:
```suggestion
                MessageBox.Show("OpenTabletDriver must be installed to the Applications folder.");
```

**Review by InfinityGhost — CHANGES_REQUESTED** [review]

**Comment by gonX** [comment]
need rebase

**Review by jamesbt365 — APPROVED** [review]

**Pull request merged**